src/elastic-search/Elasticsearch_guide.md

```diff
@@ -0,0 +1,211 @@
+| `index`                             | It contains settings that determine how an Elasticsearch index works, like how many results you can get from a search and how many copies of the data it should store. More details in [index modules](https://www.elastic.co/guide/en/elasticsearch/reference/current/index-modules.html)                                                                           |
+| `index.max_result_window`           | Determines the maximum number of documents returned in a query. `Default is 10000`                                                                                                                                                                                                                                                                                   |
+| `index.number_of_replicas`          | Sets the number of replica shards. `Default is 1`                                                                                                                                                                                                                                                                                                                    |
+| `index.mapping.total_fields.limit`  | Sets the maximum number of fields an index can have. `Default is 1000`                                                                                                                                                                                                                                                                                               |
+| `index.mapping.nested_fields.limit` | Sets the maximum number of nested fields. `Default is 50`                                                                                                                                                                                                                                                                                                            |
+| `analysis`                          | It defines how text data is processed before it is indexed, including custom analyzers and tokenizers which control the conversion of text into tokens or terms. More details in [analyzer](https://www.elastic.co/guide/en/elasticsearch/reference/current/analyzer.html) and [analysis](https://www.elastic.co/guide/en/elasticsearch/reference/8.8/analysis.html) |
+
+</br>
+
+**Dynamic mapping:**
+
+Dynamic mapping is currently only used for scientific field. More details in [Dynamic template](https://www.elastic.co/guide/en/elasticsearch/reference/8.8/dynamic-templates.html)
+
+```
+export const dynamic_template: Record<string, MappingDynamicTemplate>[] = [
```

Comment:
can this get read from a yaml/json file ?

Reply:
yes it can be read from file. Our thoughts was to implement it in a future release.